std/uni/package.d

```diff
@@ -8790,20 +8798,24 @@ inout(C)[] normalize(NormalizationForm norm=NFC, C)(return scope inout(C)[] inpu
         // reset variables
         decomposed.length = 0;
         () @trusted {
-            decomposed.assumeSafeAppend();
+            // assumeSafeAppend isn't considered pure as of writing, hence the
+            // cast. It isn't pure in the sense that the elements after
+            // the array in question are affected, but we don't use those
+            // making the call pure for our purposes.
+            (cast(void delegate() pure nothrow) {decomposed.assumeSafeAppend();})();
```

Comment:
`assumeSafeAppend`  is already `nothrow` so there's no need to also cast it to it.

Reply:
It is, but a cast needs to designate the new type as whole, including those attributes the original lambda had.

I could avoid this by using `std.traits.SetFunctionAttributes` instead but using it would be uglier IMO, besides being a needless dependency.